backend/repos/query.go

```diff
@@ -0,0 +1,142 @@
+package repos
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/honeynet/ochi/backend/entities"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type QueryRepo struct {
+	findByOwnerQuery *sqlx.Stmt
+	deleteQuery      *sqlx.Stmt
+	createQuery      *sqlx.NamedStmt
+	updateQuery      *sqlx.Stmt
+	getByIdQuery     *sqlx.Stmt
+	db               *sqlx.DB
+}
+
+func NewQueryRepo(db *sqlx.DB) (*QueryRepo, error) {
+	r := &QueryRepo{}
+	db.Mapper = reflectx.NewMapperFunc("json", strings.ToLower)
+	_, err := db.Exec(`CREATE TABLE IF NOT EXISTS queries (
+		id TEXT PRIMARY KEY NOT NULL
+		, content TEXT NOT NULL
+		, owner_id TEXT NOT NULL
+		, active INTEGER NOT NULL
+		, description TEXT NOT NULL
+		, CONSTRAINT id_unique UNIQUE (id)
+	)`)
+	if err != nil {
+		return nil, err
+	}
+	r.findByOwnerQuery, err = db.Preparex("SELECT * FROM queries WHERE owner_id=?")
+	if err != nil {
+		return nil, err
+	}
+	r.createQuery, err = db.PrepareNamed("INSERT INTO queries (id, owner_id, content, active, description) VALUES (:id, :owner_id, :content, :active, :description)")
```

Comment:
You can make multi-line strings with single quotes. Alternatively you can have the queries in separate variables.

Reply:
Done